examples/axi_lite_slave/tests/test_axi_lite_slave.py

```diff
@@ -157,8 +148,8 @@ async def write_fail(dut):
         print("Exception: %s" % str(e))
         dut._log.info("Bus successfully raised an error")
         raise TestSuccess()
-    raise TestFailure("AXI bus should have raised an error when writing to \
-                        an invalid address")
+    raise TestFailure("AXI bus should have raised an error when writing to "
+                      "an invalid address")
```

Comment:
I would also remove the `raise TestSuccess()` line above. New idiomatic is `assert` for failures, falling off the end of the test for passes. I think it was stated as a stretch goal to remove the `TestFinish` exceptions, but I can't seem to find it? @eric-wieser?

```suggestion
    assert False, "AXI bus should have raised an error when writing to an invalid address"
```

Reply:
The suggestion is fine, but currently we still need the ``raise TestSuccess()`` line to prevent this line from being reached..

Reply:
You could use an `else` clause. The entire thing could be refactored to use `assert_raises` from `test_cocotb`. If you wish to use `assert`s, I would do it all the way, even if it requires some minor refactoring.

I'm wondering if we can use pytest for that? We could start depending on pytest for test regressions. That is far less egregious than a runtime dependency. But you can ignore this off-tangent rambling for this PR.

Reply:
Indeed, having the ``assert_raises`` available generally would be useful; something for a different PR.